### PR TITLE
External References @ Attribution Tab 

### PR DESCRIPTION
**Pull request by utku-ozturk**
**First merge the PR 4dn-dcic/shared-portal-components#38 and release a new version. Then update SPC version of this PR and merge.**

**Screenshot (ExpSet Attribution Tab):**
Screen Shot 2020-08-11 at 12 29 26

**Screenshot (Experiment Attribution Tab):**
Screen Shot 2020-08-11 at 12 29 44

### CODE REVIEW

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/encoded/static/components/item-pages/components/AttributionTabView.js  2
- src/encoded/static/components/browse/components/FileEntryBlock.js  1
         

Clones added
============
- src/encoded/static/components/item-pages/components/AttributionTabView.js  3
- src/encoded/static/components/item-pages/DefaultItemView.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039&bid=19656797)

File: src/encoded/static/components/item-pages/components/AttributionTabView.js [hunk 1]
```diff
@@ -5,9 +5,13 @@ import _ from 'underscore';
...
 import ReactTooltip from 'react-tooltip';
 import { object } from '@hms-dbmi-bgm/shared-portal-components/es/components/util';
-import { ItemFooterRow } from '@hms-dbmi-bgm/shared-portal-components/es/components/ui/ItemFooterRow';
+import { ItemFooterRow, ExternalReferenceLink } from '@hms-dbmi-bgm/shared-portal-components/es/components/ui/ItemFooterRow';
+import { StackedBlockTable, StackedBlock, StackedBlockList, StackedBlockName, StackedBlockNameLabel } from '@hms-dbmi-bgm/shared-portal-components/es/components/browse/components/StackedBlockTable';
 import { FormattedInfoBlock, WrappedCollapsibleList } from './FormattedInfoBlock';
 import { Publications } from './Publications';
+import { FileEntryBlock } from '../../browse/components/FileEntryBlock';
+import { expFxn, Schemas } from './../../util';
+import { experimentsFromExperimentSet } from '../../util/experiments-transforms';
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/AttributionTabView.js:14]: Codacy found an issue: ['experimentsFromExperimentSet' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

File: src/encoded/static/components/item-pages/components/AttributionTabView.js [hunk 2]
```diff
@@ -65,9 +69,210 @@ export const ContactPersonListItem = React.memo(function ContactPersonListItem({
...
+        }
 
+        let fileTitleString;
+        if (file.accession) {
+            fileTitleString = file.accession;
+        }
+        if (!fileTitleString && fileAtId) {
+            var idParts = _.filter(fileAtId.split('/'));
+            if (idParts[1].slice(0, 5) === '4DNFI') {
+                fileTitleString = idParts[1];
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/AttributionTabView.js:123]: Codacy found an issue: [Use array destructuring.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/encoded/static/components/item-pages/components/AttributionTabView.js  2
- src/encoded/static/components/browse/components/FileEntryBlock.js  1
         

Clones added
============
- src/encoded/static/components/item-pages/components/AttributionTabView.js  3
- src/encoded/static/components/item-pages/DefaultItemView.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039&bid=19656797)

File: src/encoded/static/components/item-pages/components/AttributionTabView.js (hunk 1 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/AttributionTabView.js:14]: Codacy found an issue: ['experimentsFromExperimentSet' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

File: src/encoded/static/components/item-pages/components/AttributionTabView.js (hunk 2 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/AttributionTabView.js:123]: Codacy found an issue: [Use array destructuring.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 5
           

Complexity increasing per file
==============================
- src/encoded/static/components/browse/components/FileEntryBlock.js  1
- src/encoded/static/components/item-pages/components/ItemFooterRow.js  12
         

Clones added
============
- src/encoded/static/components/item-pages/DefaultItemView.js  2
- src/encoded/static/components/item-pages/components/ItemFooterRow.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039&bid=19656797)

File: src/encoded/static/components/item-pages/components/ItemFooterRow.js [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+        }
+
+        let fileTitleString;
+        if (file.accession) {
+            fileTitleString = file.accession;
+        }
+        if (!fileTitleString && fileAtId) {
+            var idParts = _.filter(fileAtId.split('/'));
+            if (idParts[1].slice(0, 5) === '4DNFI') {
+                fileTitleString = idParts[1];
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/ItemFooterRow.js:186]: Codacy found an issue: [Use array destructuring.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

File: src/encoded/static/components/item-pages/components/AttributionTabView.js [hunk 4]
```diff
@@ -4,11 +4,9 @@ import React from 'react';
 import _ from 'underscore';
 import PropTypes from 'prop-types';
 import ReactTooltip from 'react-tooltip';
-import { object } from '@hms-dbmi-bgm/shared-portal-components/es/components/util';
-import { ItemFooterRow } from '@hms-dbmi-bgm/shared-portal-components/es/components/ui/ItemFooterRow';
 import { FormattedInfoBlock, WrappedCollapsibleList } from './FormattedInfoBlock';
 import { Publications } from './Publications';
-
+import { ItemFooterRow, ExternalReferencesStackedTable } from './ItemFooterRow';
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/AttributionTabView.js:9]: Codacy found an issue: ['ExternalReferencesStackedTable' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

File: src/encoded/static/components/item-pages/components/ItemFooterRow.js [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+                            <li key={i}>{ altAccession }</li>
+                        );
+                    }) }
+                </ul>
+            </div>
+        </div>
+    );
+}
+
+function AliasesSection({ aliases, actions }){
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/ItemFooterRow.js:97]: Codacy found an issue: ['AliasesSection' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

File: src/encoded/static/components/item-pages/components/ItemFooterRow.js [hunk 6]
```diff
@@ -0,0 +1,315 @@
+'use strict';
+
+import React from 'react';
+import _ from 'underscore';
+import { FileEntryBlock } from '../../browse/components/FileEntryBlock';
+import { expFxn, Schemas } from './../../util';
+import { console, object } from '@hms-dbmi-bgm/shared-portal-components/es/components/util';
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/ItemFooterRow.js:7]: Codacy found an issue: ['console' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

File: src/encoded/static/components/item-pages/components/ItemFooterRow.js [hunk 7]
```diff
@@ -0,0 +1,315 @@
...
+ * Shown at bottom of Item pages.
+ *
+ * @class ItemFooterRow
+ * @type {Component}
+ * @prop {Object} context - JSON representation of current Item object. Should be available through Redux store's context.
+ * @prop {Object} schemas - JSON representation of sitewide schemas.
+ * @prop {number} width - Used to define ExternalReferencesStackedTable's width
+ */
+export const ItemFooterRow = React.memo(function ItemFooterRow({ context, schemas, width }) {
+    const { aliases = [], actions = [], external_references = [], alternate_accessions = [] } = context || {};
```
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/ItemFooterRow.js:21]: Codacy found an issue: ['aliases' is assigned a value but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

**Comment by utku-ozturk** [comment]
`ItemFooterRow.js` has been moved from SPC to FF since there is not any plans to use it in CGAP at the moment. Then `ExternalReferencesStackedTable` has also been moved into `ItemFooterRow` to group all related stuff.

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 4
           

Complexity increasing per file
==============================
- src/encoded/static/components/item-pages/components/ItemFooterRow.js  12
- src/encoded/static/components/browse/components/FileEntryBlock.js  1
         

Clones added
============
- src/encoded/static/components/item-pages/components/ItemFooterRow.js  3
- src/encoded/static/components/item-pages/DefaultItemView.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039&bid=19656797)

File: src/encoded/static/components/item-pages/components/ItemFooterRow.js (hunk 3 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/ItemFooterRow.js:186]: Codacy found an issue: [Use array destructuring.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

File: src/encoded/static/components/item-pages/components/ItemFooterRow.js (hunk 5 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/ItemFooterRow.js:97]: Codacy found an issue: ['AliasesSection' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

File: src/encoded/static/components/item-pages/components/ItemFooterRow.js (hunk 7 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/ItemFooterRow.js:21]: Codacy found an issue: ['aliases' is assigned a value but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

File: src/encoded/static/components/item-pages/components/ItemFooterRow.js (hunk 6 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/ItemFooterRow.js:7]: Codacy found an issue: ['console' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 4
           

Complexity increasing per file
==============================
- src/encoded/static/components/browse/components/FileEntryBlock.js  1
- src/encoded/static/components/item-pages/components/ItemFooterRow.js  12
         

Clones added
============
- src/encoded/static/components/item-pages/DefaultItemView.js  2
- src/encoded/static/components/item-pages/components/ItemFooterRow.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039&bid=19656797)

File: src/encoded/static/components/item-pages/components/ItemFooterRow.js (hunk 3 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/ItemFooterRow.js:186]: Codacy found an issue: [Use array destructuring.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

File: src/encoded/static/components/item-pages/components/ItemFooterRow.js (hunk 5 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/ItemFooterRow.js:97]: Codacy found an issue: ['AliasesSection' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

File: src/encoded/static/components/item-pages/components/ItemFooterRow.js (hunk 7 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/ItemFooterRow.js:21]: Codacy found an issue: ['aliases' is assigned a value but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

File: src/encoded/static/components/item-pages/components/ItemFooterRow.js (hunk 6 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/ItemFooterRow.js:7]: Codacy found an issue: ['console' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 4
           

Complexity increasing per file
==============================
- src/encoded/static/components/browse/components/FileEntryBlock.js  1
- src/encoded/static/components/item-pages/components/ItemFooterRow.js  12
         

Clones added
============
- src/encoded/static/components/item-pages/components/ItemFooterRow.js  3
- src/encoded/static/components/item-pages/DefaultItemView.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039&bid=19656797)

File: src/encoded/static/components/item-pages/components/ItemFooterRow.js (hunk 3 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/ItemFooterRow.js:186]: Codacy found an issue: [Use array destructuring.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

File: src/encoded/static/components/item-pages/components/ItemFooterRow.js (hunk 5 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/ItemFooterRow.js:97]: Codacy found an issue: ['AliasesSection' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

File: src/encoded/static/components/item-pages/components/ItemFooterRow.js (hunk 7 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/ItemFooterRow.js:21]: Codacy found an issue: ['aliases' is assigned a value but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

File: src/encoded/static/components/item-pages/components/ItemFooterRow.js (hunk 6 above)
Comment (j1z0) [inline src/encoded/static/components/item-pages/components/ItemFooterRow.js:7]: Codacy found an issue: ['console' is defined but never used.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5995039)

**Pull request merged**